garnett/posfilewriter.py

```diff
@@ -95,17 +95,10 @@ def _write(msg, end='\n'):
 
             # shape defs
             try:
-                required = set(frame.types).intersection(
-                    set(frame.shapedef.keys()))
-                not_defined = set(frame.types).difference(
-                    set(frame.shapedef.keys()))
-                for name in required:
-                    _write('def {} "{}"'.format(name, frame.shapedef[name].pos_string))
-                for name in not_defined:
-                    logger.info(
-                        "No shape defined for '{}'. "
-                        "Using fallback definition.".format(name))
-                    _write('def {} "{}"'.format(name, DEFAULT_SHAPE_DEFINITION.pos_string))
+                if len(frame.types) != len(frame.type_shapes):
```

Comment:
I'm OK with erroring out if there are not enough shape definitions but it should probably be mentioned in the `changelog` since its a change from the expected behavior.